sql/analyzer/indexes.go

```diff
@@ -249,7 +249,7 @@ func getIndexes(
 			// either return a merged index lookup for both values, or for neither. Returning either one leads to incorrect
 			// results.
 			if !canMergeIndexLookups(result, indexes) {
-				return nil, nil
+				continue
```

Comment:
@zachmu i think this fixes the static lookup case, will add more tests to verify the parent filter is always left untouched

Reply:
Still think we need at test with <> as part of the actual join condition, e.g. `a.col <> b.col`